scripts/quests/sandoria/Distant_Loyalties.lua

```diff
@@ -0,0 +1,169 @@
+    fameArea = xi.quest.fame_area.SANDORIA,
+    item = xi.items.WHITE_CAPE,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and
+            player:getFameLevel(xi.quest.fame_area.SANDORIA) >= 4
+        end,
+
+        [xi.zone.SOUTHERN_SAN_DORIA] =
+        {
+            ['Femitte'] = quest:progressEvent(663),
+
+            onEventFinish =
+            {
+                [663] = function(player, csid, option, npc)
+                    if option == 0 then
+                        quest:begin(player)
+                        npcUtil.giveKeyItem(player, xi.ki.GOLDSMITHING_ORDER)
+                        quest:setVar(player, 'Prog', 1)
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.SOUTHERN_SAN_DORIA] =
+        {
+            ['Femitte'] =
+            {
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if questProgress == 1 then
+                        return quest:event(664)
+                    elseif
+                        questProgress == 4 and
+                        player:hasKeyItem(xi.ki.MYTHRIL_HEARTS)
+                    then
+                        return quest:progressEvent(665)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [665] = function(player, csid, option, npc)
+                    if quest:complete(player) then
+                        quest:setVar(player, 'finalCS', 1)
+                        player:delKeyItem(xi.ki.MYTHRIL_HEARTS)
+                    end
+                end,
+            },
+        },
+
+        [xi.zone.BASTOK_MARKETS] =
+        {
+            ['Michea'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if
+                        npcUtil.tradeHasExactly(trade, { { xi.items.MYTHRIL_INGOT, 1 } }) and
+                        quest:getVar(player, 'Prog') == 2
+                    then
+                        return quest:progressEvent(317)
+                    end
+                end,
+
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if
+                        questProgress == 1 and
+                        player:hasKeyItem(xi.ki.GOLDSMITHING_ORDER)
+                    then
+                        return quest:progressEvent(315)
+                    elseif questProgress == 2 then
+                        return quest:event(316)
+                    elseif
+                        questProgress == 3 and
+                        player:needToZone()
+                    then
+                        return quest:progressEvent(320)
```

Comment:
You have a progressEvent(320) but no corresponding onEventFinish below

Reply:
Also the statement below this one is identical. Something weird's going on here lol.

Reply:
There's no event 320 in the original code. so I'm guessing this isn't needed

Reply:
maybe a typo that never met a check so didnt see it ha